components/UserObserver.php

```diff
@@ -67,31 +74,129 @@ public function updateUser($params)
 						}
 					}
 
-					$contact->nick_name = $user->first_name;
-					$contact->title = $user->title;
-					$contact->first_name = $user->first_name;
-					$contact->last_name = $user->last_name;
-					$contact->qualifications = $user->qualifications;
-
-					if (!$contact->save()) {
-						throw new Exception('Unable to save contact: '.print_r($contact->getErrors(),true));
-					}
+					$contact = $this->saveContact($user, $contact);
 
 					if ($user->contact_id != $contact->id) {
 						$user->contact_id = $contact->id;
 
 						if (!$user->save()) {
+							\Yii::log("Unable to save user contact: ".print_r($user->getErrors(),true), \CLogger::LEVEL_ERROR);
 							throw new Exception("Unable to save user contact: ".print_r($user->getErrors(),true));
 						}
 					}
+					return $user;
+				} else {
+					\Yii::log("User " . $username . " not found in the CSD database.", \CLogger::LEVEL_ERROR);
+					throw new Exception("Unable to save user contact: ".$username);
 				}
-			} catch (Exception $e) {
-				// silently return back to UserIdentity without having refreshed the user
-			}
+			/*} catch (Exception $e) {
+				\Yii::log("Unable to update user. Error: ". $e->getMessage(), \CLogger::LEVEL_ERROR);
+			}*/
+		//}
+	}
+
+
+	/**
+     * Finds User by username
+     *
+     * @param string $username
+     * @return User
+     */
+	private function getUser(string $username): User
+	{
+		$criteria = new \CDbCriteria();
+		$criteria->join = 'JOIN user_authentication ua ON t.id = ua.user_id';
+		$criteria->addCondition('ua.username = :username');
+		$criteria->params[':username'] = $username;
```

Comment:
should we take institution id into account here?

Reply:
I've added institution